architecture/ambient/ztunnel.md

```diff
@@ -0,0 +1,236 @@
+Ztunnel will look up the destination from the [addresses](#address-type) it is configured with.
+
+For traffic to unknown addresses, or to workloads that are not a part of the mesh, the traffic will just be passed through as is.
+To make ztunnel more transparent, the original source IP address will be spoofed.
+Additionally, `splice` will be used to make this proxying more efficient when possible.
+
+For traffic in the mesh, things are a bit more complex:
+
+1. If the destination has a waypoint proxy, we must send to it to the waypoint (using HBONE).
+   When we do this, we will want to preserve the original destination Service IP, as the waypoint can do a better job picking a backend pod than we can.
+   Note: the application itself may have already resolved the Service IP to a specific pod if it has Kubernetes native routing built in; since we don't have the Service information in this case we will use the destination IP we received (a pod). Most notably, sidecar proxies behave this way.
+1. If the destination is on our node, we "fast path" the request and convert this into an inbound request.
+   This has the same semantics as if we had sent a request back to ourselves, but is more efficient and reduces complexity in the Ztunnel.
+1. Otherwise, we forward the request to the destination using HBONE. If the destination is a Service, we resolve this to a specific pod IP.
+
+In all cases, we spoof the original source IP.
+
+### Inbound Passthrough
+
+Traffic entering a pod that is not transmitted over HBONE  (i.e. with a destination port != 15008) is handled by the "inbound passthrough" code path, on ztunnel's port 15006.
+
+This is fairly straightforward.
+
+First, we need to check that this traffic is allowed.
+Traffic may be denied by RBAC policies (especially from a `STRICT` mode enforcement, which denies plaintext traffic).
+
+If it is allowed, we will forward to the target destination.
+
+#### Hairpin
+
+In the case that the destination has a waypoint, that waypoint must have been bypassed to reach the inbound passthrough codepath.
+How we handle this is [under discussion](https://docs.google.com/document/d/1uM1c3zzoehiijh1ZpZuJ1-SzuVVupenv8r5yuCaFshs/edit#heading=h.dwbqvwmg6ud3).
+
+### Inbound
+
+Traffic entering a pod over HBONE will be handled by the "inbound" code path, on port 15008.
+
+Incoming requests have multiple "layers": TLS wrapping HTTP CONNECT wrapping the users connection.
+
+To unwrap the first layer, we terminate TLS.
+As part of this, we need to pick the correct certificate to serve on behalf of the destination workload.
+As discussed in [HBONE](#hbone), this is based on the destination IP.
+Additionally, we enforce the peer has a valid mesh identity (but do not assert _which_ identity, yet).
+
+Next, we terminate the CONNECT.
+From the [headers](#headers), we know the target destination.
+If the target destination has a waypoint, we enforce the request is coming from that waypoint. Otherwise, the request is rejected.
```

Comment:
```suggestion
If the target destination has a waypoint, we enforce that the request is coming from that waypoint. Otherwise, the request is rejected.
```